### ISSUE TITLE
Incorrect SE calculation using IVW_MRE

### ISSUE THREAD

**Issue by rheuzhas** [MRCIEU/TwoSampleMR#381]
Hi Gib, the package calculates IVW_MRE confidence intervals and p-values that are **smaller** than IVW(-FE) when using a small number of instruments, eg, for cis-pQTL work. This seems implausible, please can you comment? Thanks in advance.

**Comment by mightyphil2000** [comment]
This is an indication that the residual standard error in the IVW model is less than 1, indicating that that there is underdispersion of the genetic effects (the opposite of heterogeneity). In a multiplicative random effects model the variance is equal to the variance for the fixed effect multiplied by the residual standard error. Usually the residual standard is greater than 1 (i.e. indicating heterogeneity) and thus the variance for the random effect is inflated (thus taking into account the heterogeneity in the estimates). In your case it's likely that the residual standard error is less than 1, and therefore the variance for the random effect will be less than that observed for the fixed effect. In these formulas below the variance for the multiplicative random effect has distribution phi^2 * variance_for_fixed_effect.

[cid:1ee8c68f-2347-488b-b7a4-c6c5e5e2de0d]

Phi is the measure of heterogeneity and is equivalent to sigma (phi =  Q/DF), where Q is Cochran's Q for heterogeneity and df is degrees of freedom.


You can estimate your residual standard error using this script:

ivw.res <- summary(lm(b_out ~ -1 + b_exp, weights = 1/se_out^2))

ivw.res$sigma


where b_out is the SNP effect on the outcome and b_exp is the SNP effect on the exposure and se_out is the standard error for b_out.


Check to see if sigma is less than 1. If it is, then you should use the fixed effects IVW model. It may also be worth considering whether you have included strongly correlated SNPs in your analysis (likely if you are conducting a cis pQTL analysis with multiple SNPs). If so, you should probably consider pruning the SNPs for independence.


Below are links to some papers that discuss this issue further.

https://link.springer.com/article/10.1007/s10654-017-0255-x
https://arxiv.org/pdf/1512.04486.pdf

Some key paragraphs copied and pasted from the above links:

Burgess and Thompson 2017:
"if the estimate of the residual standard error is less than 1, then this is not plausible (for uncorrelated variants), as there is no biological mechanism that would lead to underdispersion of the genetic associations. Therefore standard errors of the regression coefficients should be corrected by dividing by the minimum of the residual standard error estimate and one, so that a random-effects model can never give more precise estimates than a fixed-effect model. Underdispersion may be a sign that the genetic variants may have been chosen inappropriately in a way that preferentially selects similar variants. However, it may simply be a chance finding."


Burgess & Bowden 2015:
"In a multiplicative random-effects model, the βˆ IV j estimates are assumed to be normally distributed with mean β and variance φ 2 Mσ 2 IV j . This model can be fitted by linear regression of the βˆ Y j on the βˆXj using the σ −2 Y j as weights. A fixed-effect model can be fitted by setting the residual standard error in the regression model to be one; this can be achieved after fitting the regression model by dividing the standard error by the estimate of the residual standard error (Thompson and Sharp, 1999). A multiplicative random-effects model can be fitted by allowing the residual standard error (which is equivalent to the heterogeneity parameter φM ) to be estimated as part of the model. The multiplicative random-effects model is therefore equivalent to an overdispersed regression model. In case of underdispersion (that is, the estimated residual standard error is less than one), the standard errors should be fixed by setting φˆM = 1, as any underdispersion is assumed to occur by chance, and not to be empirically justified."





________________________________
From: rheuzhas ***@***.***>
Sent: 28 September 2022 14:56
To: MRCIEU/TwoSampleMR ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [MRCIEU/TwoSampleMR] Incorrect SE calculation using IVW_MRE (Issue #381)


Hi Gib, the package calculates IVW_MRE confidence intervals and p-values that are smaller than IVW(-FE) when using a small number of instruments, eg, for cis-pQTL work. This seems implausible, please can you comment? Thanks in advance.

**Comment by rheuzhas** [comment]
Thank you very much Phil. Please may I clarify as a non-methodologist:

1) When there is underdispersion* (as you say, opposite of heterogeneity), might it make more sense to use FE-IVW a priori? Since MRE is "designed" to (among other things) account for presence of heterogeneity.

*one can anticipate this in cis-pQTL work with "minimally correlated" instruments (r2<0.1)

2) How does one operationalise this in the 2 sample setting? Because sqrt( (se^2)*phi^2 ) still provides an ultimately smaller se. (eg, Q=4.14, DF=9, giving phi<1)

**Comment by mightyphil2000** [comment]
Yes when there is under-dispersion it is better to use a fixed effects model.

An r2 of 0.1 is actually quite a strong correlation for two sample MR using summary data. That is probably why you are getting under-dispersion. Q=4.14, DF=9 indicates that your results are very similar amongst the different pQTLs. I would use a much stricter r2 threshold or model the correlation structure amongst your SNPs using a correlation matrix.


________________________________
